keel-sql/src/main/resources/db/changelog/20200522-standardize-timestamps.yml

```diff
@@ -0,0 +1,23 @@
+databaseChangeLog:
+- changeSet:
+    id: event-uid-timestamp-index
+    author: fletch
+    changes:
+    - sql:
+        sql: |
+          alter table agent_lock modify expiry timestamp(3) not null;
+          alter table artifact_last_checked modify at timestamp(3) not null;
+          alter table delivery_config_last_checked modify at timestamp(3) not null;
+          alter table diff_fingerprint modify first_detection_time timestamp(3) not null;
+          alter table environment_artifact_constraint modify created_at timestamp(3) not null;
+          alter table environment_artifact_constraint modify judged_at timestamp(3) null;
+          alter table environment_artifact_pin modify pinned_at timestamp(3) not null;
+          alter table environment_artifact_queued_approval modify queued_at timestamp(3) not null;
+          alter table environment_artifact_versions modify approved_at timestamp(3) null;
+          alter table environment_artifact_versions modify deployed_at timestamp(3) null;
+          alter table environment_artifact_versions modify replaced_at timestamp(3) null;
+          alter table event modify timestamp timestamp(3) not null;
+          alter table paused modify paused_at timestamp(3) null;
+          alter table resource_last_checked modify at timestamp(3) not null;
+          alter table task_tracking modify timestamp timestamp(3) not null;
+          alter table unhappy_veto modify recheck_time timestamp(3) not null;
```

Comment:
done with `sql` rather than `modifyDataType` as the latter does not preserve `null` / `not null` (and other column. settings) on MySQL.